### PR TITLE
Add S3RandomAccessFile

### PR DESCRIPTION
**Pull request by sduzh**
## Problem Summary(Required) ：

Add S3RandomAccessFile for accessing S3 objects.

## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] others

### CODE REVIEW

**Review by dirtysalt — DISMISSED** [review]

**Comment by sduzh** [comment]
@wyb PTAL

**Comment by sduzh** [comment]
@starrocks-xueli PTAL

File: be/src/io/input_stream.h [hunk 1]
```diff
@@ -21,7 +21,7 @@ class InputStream : public Readable {
     virtual Status skip(int64_t count) = 0;
 
     // Returns true if InputStream support `peak()`
-    virtual bool allows_peak() const = 0;
+    virtual bool allows_peak() const { return false; }
```
Comment (starrocks-xueli) [inline be/src/io/input_stream.h:24]: is there any upper class will call this method? maybe we don't need `peak`?

File: be/src/io/s3_random_access_file.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        ASSIGN_OR_RETURN(_size, S3RandomAccessFile::get_size());
+    }
+    if (offset < 0) {
+        return Status::InvalidArgument(fmt::format("Invalid offset {}", offset));
+    }
+    if (offset >= _size) {
+        return 0;
+    }
+    auto range = fmt::format("bytes={}-{}", offset, std::min<int64_t>(offset + count, _size));
+    Aws::S3::Model::GetObjectRequest request;
```
Comment (starrocks-xueli) [inline be/src/io/s3_random_access_file.cpp:32]: why not move this call into `S3Client`？

Comment (starrocks-xueli) [reply be/src/io/s3_random_access_file.cpp:32]: there is already a `S3Client` in StarRocks, why not use that?

File: be/src/io/input_stream.h (hunk 1 above)
Comment (sduzh) [reply be/src/io/input_stream.h:24]: Not yet, but there will be.

File: be/src/io/s3_random_access_file.cpp (hunk 2 above)
Comment (sduzh) [reply be/src/io/s3_random_access_file.cpp:32]: 'IO' module was designed to be the most fundamental utility, so I tend to make it independent from others.
Make others depend on 'IO', not reversed.

**Review by dirtysalt — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Pull request merged**